### PR TITLE
feat: allow regular expressions to be passed to search

### LINKED ISSUE #105: Feature: allow search to take in a RegExp
Change `emoji.search()` to return emoji whose name _contains_ the search string.  I can imagine two approaches:

1. Always treat the argument as a substring. E.g., `emoji.search('evil')` → `[{ key: 'see_no_evil', emoji: '🙈' }, { key: 'see_no_evil', emoji: '🙈' }, { key: 'see_no_evil', emoji: '🙈' }]`. (Example results not exhaustive.)
	* This is probably easiest and most lightweight to implement, but is not backwards compatible with old code that uses this library. 
2. Allow standard regular expressions in the argument. E.g., `emoji.search('.*evil')` → `[{ key: 'see_no_evil', emoji: '🙈' }, { key: 'see_no_evil', emoji: '🙈' }, { key: 'see_no_evil', emoji: '🙈' }]`. (Example results not exhaustive.)
	  * This may be more complicated, but it will be backwards compatible and it won't return unexpected results.

### PR DESCRIPTION
**Pull request by ikelax**
## PR Checklist

- [x] Addresses an existing open issue: fixes #105
- [x] That issue was marked as [`status: accepting prs`](https://github.com/omnidan/node-emoji/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/omnidan/node-emoji/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

`search` accepts `RegExp` objects as an argument. They are evaluated for each emoji.

💖

### CODE REVIEW

File: src/search.ts [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
 import { emojiData } from './data.js'
 import { normalizeName } from './utils.js'
 
-export const search = (keyword: string) => {
-  assert.string(keyword)
+/**
+ * Search for emojis containing the provided name or pattern in their name.
+ */
+export const search = (keyword: RegExp | string) => {
+  assert.any([is.default.string, is.default.regExp], keyword)
```
Comment (JoshuaKGoldberg) [inline src/search.ts:10]: [Non-actionable] Hmm, it is unfortunate to have to use the `.default` before every property of `is`. But we're on an older version of `@sindresorhus/is` that doesn't set up ESM defaults properly yet. I think this is reasonable for now, and a followup issue could be to bump `@sindresorhus/is` to latest.

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great to me, nicely done! 🔥

**Comment by JoshuaKGoldberg** [comment]
We can ignore the `lint` CI failures - I just fixed them on the `main` branch.

File: src/search.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/search.ts:10]: Oh I mean I'm not asking you to take any action in this PR. I'm not requesting changes, just noting something for the future.

**Review by omnidan — APPROVED** [review]
Thanks for the PR @ikelax! And thanks @JoshuaKGoldberg for the review ☺️ looks good to me too!

**Pull request merged**